### PR TITLE
improve -Xdev help text

### PR DESCRIPTION
**Pull request by martijnhoekstra**
Make it clear that Xdev intended for scalac development -- we're all developers here.

### CODE REVIEW

**Comment by martijnhoekstra** [comment]
https://gitter.im/scala/scala?at=5d5e58b9c508da44c35f68cb for motivation

File: src/compiler/scala/tools/nsc/settings/ScalaSettings.scala [hunk 1]
```diff
@@ -110,7 +110,7 @@ trait ScalaSettings extends StandardScalaSettings with Warnings {
    */
   val Xhelp              = BooleanSetting      ("-X", "Print a synopsis of advanced options.")
   val checkInit          = BooleanSetting      ("-Xcheckinit", "Wrap field accessors to throw an exception on uninitialized access.")
-  val developer          = BooleanSetting      ("-Xdev", "Indicates user is a developer - issue warnings about anything which seems amiss")
+  val developer          = BooleanSetting      ("-Xdev", "Issue warnings about anything which seems amiss in compiler internals, intended for compiler developers")
```
Comment (hrhino) [inline src/compiler/scala/tools/nsc/settings/ScalaSettings.scala]: (comma splice)

**Review by hrhino — APPROVED** [review]
I can't wait to see Seth's "You Are A Developer" talk.

**Review by hrhino — APPROVED** [review]
Thanks! I had decided against complaining and removed the comment.

**Pull request merged**